### PR TITLE
DHFPROD-2135: FE Implementation Display MLCP command in ingest step

### PR DESCRIPTION
**Pull request by xnikhil08**
1. Updated the MLCP command with suggested changes related to step number and CSS.
2. Rebased the repository to resolve conflicts.

@wooldridge - Please review.

### CODE REVIEW

File: web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts [hunk 1]
```diff
@@ -1,16 +1,18 @@
-import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
```
Comment (wooldridge) [inline web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts]: The ViewChild was added as part of the following: https://github.com/marklogic/marklogic-data-hub/pull/2689
I don't think it should be removed.

File: web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts [hunk 2]
```diff
@@ -33,18 +33,17 @@ import * as _ from "lodash";
     (stepSelected)="stepSelected($event)"
     (stepUpdate)="updateStep($event)"
     (stepDelete)="deleteStep($event)"
+    (clipboardSuccess)="clipboardSuccess.emit($event)"
   ></app-edit-flow-ui>
 `
 })
 export class EditFlowComponent implements OnInit, OnDestroy {
-
-  @ViewChild(EditFlowUiComponent) editFlowUi: EditFlowUiComponent;
```
Comment (wooldridge) [inline web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts]: The ViewChild code was added as part of the following: https://github.com/marklogic/marklogic-data-hub/pull/2689
I don't think it should be removed.

File: web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts [hunk 3]
```diff
@@ -180,7 +174,6 @@ export class EditFlowComponent implements OnInit, OnDestroy {
   }
   updateStep(step) {
     this.manageFlowsService.updateStep(this.flow.id, step.id, step).subscribe(resp => {
-      this.editFlowUi.stepUpdated(step);
```
Comment (wooldridge) [inline web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts]: The stepUpdated() was added as part of the following: https://github.com/marklogic/marklogic-data-hub/pull/2689
I don't think it should be removed.

File: web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts [hunk 4]
```diff
@@ -23,8 +25,6 @@ import * as _ from "lodash";
     [selectedStepId]="selectedStepId"
     [projectDirectory]="projectDirectory"
     [flowEnded]="flowEnded"
-    [runFlowClicked]="runFlowClicked"
```
Comment (wooldridge) [inline web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts]: @brucean52 was this (and maybe other changes) part of DHFPROD-2627?

File: web/src/main/ui/app/components/flows-new/edit-flow/ingest/ingest.component.ts [hunk 5]
```diff
@@ -37,7 +47,6 @@ export class IngestComponent implements OnInit {
     } = this.step.fileLocations;
 
     const {
-      additionalCollections,
```
Comment (wooldridge) [inline web/src/main/ui/app/components/flows-new/edit-flow/ingest/ingest.component.ts]: @brucean52 was this part of DHFPROD-2561? Want to make sure working code isn't deleted with this PR.

**Comment by aebadirad** [comment]
This branch needs a rebase with develop. Could we get a rebase?

**Comment by xnikhil08** [comment]
I rebased the this branch with develop and implemented all the suggested changes.

@aebadirad & @wooldridge - Please review now and let me know if you still find any issues.

**Comment by xnikhil08** [comment]
> I'm testing this feature with the URI replacement information in the documentation here:
> 
> https://docs.marklogic.com/guide/mlcp/import
> 
> under "Transforming the Default URI".
> 
> When I type the following in the Target URI Replacement field:
> 
> `/space,'',/bill/data/,'/will/'`
> 
> I see this in the MLCP command:
> 
> `-output_uri_replace "/space,'',/bill/data/,'/will',''"`
> 
> It appears there is an extra '' adde to the comma-separated list. According to the documentation it should look like this:
> 
> `-output_uri_replace "/space,'',/bill/data/,'/will/'"`

@wooldridge - It is fixed. Please review now.

**Comment by xnikhil08** [comment]
> I'm testing the step index information that appears in the command. When I have a single Ingestion step, it looks good, we have:
> 
> step=0
> 
> But when I add additional Ingestion steps an select those, the command shows the following:
> 
> step=-1
> 
> See my screencast:
> 
> https://drive.google.com/file/d/1tb0xkRDywo56lacsVbC-zK-IieDEqo1l/view
> 
> Also, I'm seeing the step indexes in the flow artifacts (the on-disk files) are 1-based, not 0-based.

@wooldridge - Fixed these issues and committed the changes. Please review now./

**Comment by srinathgit** [comment]
@xnikhil08 ,
If no value is set for an option, we don't have to include that in the mlcp command. For example, Target URI Replacement is not set in this case and we don't have to include -output_uri_replace option in the mlcp command

image

File: web/src/main/ui/app/components/flows-new/edit-flow/ingest/ui/ingest-ui.component.ts [hunk 6]
```diff
@@ -127,7 +151,82 @@ export class IngestUiComponent implements OnInit{
...
+    this.mlcp = {};
+    this.addMlcpOption(options, 'import', null, false, false);
+    this.addMlcpOption(options, 'mode', 'local', false, true);
+
+    const host = this.envService.settings.host;
+    const port = this.envService.settings.stagingPort;
+    const username = this.envService.settings.mlUsername;
+    let input_file_path = this.step.fileLocations.inputFilePath;
+    let input_file_type = this.step.fileLocations.inputFileType;
+    let document_type = this.step.options.outputFormat.toLowerCase();
```
Comment (srinathgit) [inline web/src/main/ui/app/components/flows-new/edit-flow/ingest/ui/ingest-ui.component.ts]: The value for input_file_type for mlcp is not the same as the one chosen from the dropdown. Please check the link https://docs.marklogic.com/guide/mlcp/import#id_71112

Comment (xnikhil08) [reply web/src/main/ui/app/components/flows-new/edit-flow/ingest/ui/ingest-ui.component.ts]: This is Fixed. Please review now.

**Comment by xnikhil08** [comment]
> @xnikhil08 when I create a new ingestion step in my flow and select it, I'm seeing step=0 in the MLCP command for that step:
> 
> https://drive.google.com/file/d/1twVcX4zo2Na0cRq07ieVCIn7DZ2Qmxou/view
> 
> In the above case, it should read: step=2

@wooldridge :- Implemented the changes and fixed all the issues. Please review now.

**Pull request merged**